client/src/components/Messaging/ScrollWrapper.tsx

```diff
@@ -8,48 +8,52 @@ type ScrollWrapperProps = {
 
 export const ScrollWrapper = ({ children, messageCount }: ScrollWrapperProps) => {
   const wrapperRef = useRef<HTMLDivElement>(null);
+  const [isAtBottom, setIsAtBottom] = useState(true);
   const [showPopup, setShowPopup] = useState(false);
-  const [scrollAtBottom, setScrollAtBottom] = useState(true);
 
   useEffect(() => {
-    if (wrapperRef.current) {
+    const handleScroll = () => {
       const wrapper = wrapperRef.current;
-      const isScrolledToBottom = wrapper.scrollHeight - wrapper.scrollTop === wrapper.clientHeight;
-      setScrollAtBottom(isScrolledToBottom);
-      if (isScrolledToBottom) {
-        wrapper.scrollTop = wrapper.scrollHeight;
-      } else {
-        setShowPopup(true);
+      if (!wrapper) return;
+      setIsAtBottom(wrapper.scrollTop + wrapper.clientHeight >= wrapper.scrollHeight);
+    };
+
+ const wrapperElement = wrapperRef.current;
+    if (wrapperElement) {
+      wrapperElement.addEventListener('scroll', handleScroll);
+    }
+
+    return () => {
+      if (wrapperElement) {
+        wrapperElement.removeEventListener('scroll', handleScroll);
       }
+    };
+  }, []);
+
+  useEffect(() => {
+    if (isAtBottom && messageCount > 0) {
+      scrollToBottom();
     }
-  }, [messageCount]);
+  }, [messageCount, isAtBottom]);
 
   useEffect(() => {
-    if (scrollAtBottom && wrapperRef.current) {
-      wrapperRef.current.scrollTop = wrapperRef.current.scrollHeight;
+    if (!isAtBottom && messageCount > 0) {
+      setShowPopup(true);
     }
-  }, [scrollAtBottom, messageCount]);
+// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why do we need it? `isAtBottom` should be added to hook-dependency.

Reply:
I optimized the code, but I didn’t make modifications here because I only want to perform the check when the message count changes. The condition for the check will use isAtBottom. If isAtBottom is added to the dependency array, it will trigger the check whenever the user scrolls up in the chat box, causing the popup to appear.